### PR TITLE
Fix logging on geochaching.com

### PR DESCRIPTION
**Pull request by jgoestl**
Add payload as json and use cookie authentication

### CODE REVIEW

**Comment by jgoestl** [comment]
Fixing issue #8154

**Comment by Lineflyer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=5262679&bid=17241226)

File: main/src/cgeo/geocaching/connector/gc/GCWebAPI.java [hunk 1]
```diff
@@ -389,7 +443,13 @@ static MapSearchResultSet searchMap(@NonNull final Viewport viewport) {
                     put("logText", logInfo).
                     put("usedFavoritePoint", formatBoolean(addToFavorites));
 
-            final PostLogResponse response = postAPI("/web/v1/geocache/" + StringUtils.lowerCase(geocache.getGeocode()) + "/GeocacheLog", params, PostLogResponse.class).blockingGet();
+            GeocacheLog geocacheLog = new GeocacheLog(geocache.getCacheId(), geocache.getGeocode(),
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/connector/gc/GCWebAPI.java]: Codacy Issue found: [Variable 'geocacheLog' should be declared final.](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=5262679)

File: main/src/cgeo/geocaching/network/Network.java [hunk 2]
```diff
@@ -223,6 +223,15 @@ private Network() {
         return RxOkHttpUtils.request(OK_HTTP_CLIENT, request.build());
     }
 
+    @NonNull
+    public static <T> Single<T> postJsonRequest(final String uri, final Class<T> clazz, final Object jsonObject) throws JsonProcessingException {
+        final Builder request = new Builder().url(uri).post(RequestBody.create(MEDIA_TYPE_APPLICATION_JSON,
+                mapper.writeValueAsString(jsonObject)));
+        Single<Response> response = RxOkHttpUtils.request(OK_HTTP_CLIENT, request.build());
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/network/Network.java]: Codacy Issue found: [Variable 'response' should be declared final.](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=5262679)

**Comment by Lineflyer** [comment]
OK to test (for CI)

**Comment by moving-bits** [comment]
retest this please (for CI)

**Comment by moving-bits** [comment]
Hm, still fails with:
```
* What went wrong:
21:57:08  Execution failed for task ':main:installBasicDebug'.
21:57:08  > com.android.builder.testing.api.DeviceException: Could not create ADB Bridge. ADB location: /opt/android-sdk-linux/platform-tools/adb
```
Will try again once.

**Comment by moving-bits** [comment]
retest this please

**Comment by Lineflyer** [comment]
@jgoestl 

Just a minor issue left accoring to our CI:

GCWebAPI.java:33, ImportOrderCheck, Priorität: Niedrig
Extra separation in import group before 'io.reactivex.Single'
Checks the ordering/grouping of imports

**Comment by Lineflyer** [comment]
@pstorch May I invite you for a review as original author of that code part?

**Comment by Lineflyer** [comment]
retest this please

**Comment by moving-bits** [comment]
Those two tests consistently fail:

cgeo.geocaching.address.GeocoderTest.testAndroidGeocoder
cgeo.geocaching.CgeoApplicationTest.testSearchByViewport

@jgoestl 
Can you cross-check locally, please, if those tests succeed in your dev environment?

**Comment by jgoestl** [comment]
@moving-bits 
I cross-checked it locally and the tests succeeded

File: main/src/cgeo/geocaching/connector/gc/GCWebAPI.java [hunk 3]
```diff
@@ -232,6 +233,13 @@ String getAuthorizationField() {
         @JsonProperty("longitude")
         double longitude;
 
+        PostedCoordinates(final double latitude, final double longitude) {
+            this.latitude = latitude;
+            this.longitude = longitude;
+        }
+
+        PostedCoordinates() {}
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/connector/gc/GCWebAPI.java]: Codacy Issue found: ['}' is not preceded with whitespace.](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=5262679)

**Comment by Lineflyer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=5262679&bid=17241226)

File: main/src/cgeo/geocaching/connector/gc/GCWebAPI.java [hunk 4]
```diff
@@ -240,6 +248,12 @@ Geopoint toCoords() {
     @JsonIgnoreProperties(ignoreUnknown = true)
     static final class CallerSpecific {
         boolean favorited;
+
+        CallerSpecific(final boolean favorited) {
+            this.favorited = favorited;
+        }
+
+        CallerSpecific() {}
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/connector/gc/GCWebAPI.java]: Codacy Issue found: ['}' is not preceded with whitespace.](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=5262679)

**Comment by moving-bits** [comment]
> I cross-checked it locally and cgeo.geocaching.address.GeocoderTest.testAndroidGeocoder succeeded.
> The other test is now fixed, I added missing default constructors

@jgoestl Thanks for checking and for fixing the other test.

**Comment by pstorch** [comment]
> @pstorch May I invite you for a review as original author of that code part?

Sorry, I don't have the dev environment prepared for c:geo anymore. But a quick glance at the change looks good to me. Thanks to the new contributors. 👍

**Comment by Lineflyer** [comment]
Thanks @pstorch for taking a look at it.

I will merge this to `release´ then merge branches, and build RC and new nightly. So we have the change in all development version.

**Comment by Lineflyer** [comment]
Thank @jgoestl for your helpful contribution!

**Comment by jgoestl** [comment]
my pleasure :blush:

**Comment by Lineflyer** [comment]
> my pleasure 😊

Could you maybe also take a look at the remaining issue I found --> #8166

**Pull request merged**